lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -238,8 +238,23 @@ def _get_sans(self, cert_type: CertType) -> Dict[str, List[str]]:
         if cert_type == CertType.APP_ADMIN:
             return sans
 
-        sans["sans_ip"] = [self.charm.unit_ip]
-        sans["sans_dns"] = [self.charm.unit_name, socket.getfqdn()]
+        dns = {self.charm.unit_name, socket.gethostname(), socket.getfqdn()}
+        ips = {self.charm.unit_ip}
+        if cert_type == CertType.UNIT_HTTP and self.charm.unit_public_ip:
+            ips.add(self.charm.unit_public_ip)
+
+        for ip in ips.copy():
+            try:
+                name, aliases, addresses = socket.gethostbyaddr(ip)
+                ips.update(addresses)
+
+                dns.add(name)
```

Comment:
fqdn and back resolve hostname can be the same.
```suggestion
                dns.update(name)
```